app/models/aaib_report.rb

```diff
@@ -0,0 +1,33 @@
+class AaibReport < Document
+
+  validates :date_of_occurrence, presence: true, date: true, unless: ->(report) {
+    report.report_type == "safety-study" && report.date_of_occurrence.blank?
+  }
+
+  FORMAT_SPECIFIC_FIELDS = [
+    :date_of_occurrence,
+    :aircraft_category,
+    :report_type,
+    :location,
+    :aircraft_type,
+    :registration,
+  ]
+
+  attr_accessor *FORMAT_SPECIFIC_FIELDS
+
+  def initialize(params = {})
+    super(params, FORMAT_SPECIFIC_FIELDS)
```

Comment:
I still need to set them as `attr_accessor`s. Can I do that with your solution or is that something you missed?